SLE 2024-07 release notes (#208)
ni/install-systemlink-enterprise: release-notes/2024-07/README.md

# SystemLink Enterprise release 2024-07 Release Notes

The 2024-07 release bundle for SystemLink Enterprise has been published to <https://downloads.artifacts.ni.com>. This update includes new features, bug fixes, and security updates. Work with your account representative to obtain credentials to access these artifacts. If you are not upgrading from the previous release, refer to past release notes to ensure you have addressed all required configuration changes.

## New Features and Behavior changes

- Behavior change or new feature description
- `helium-serviceregistry 0.16.7`
    - The Alarm Service privileges are exposed under the `Alarms` category in the security UI.
- `testmonitorservice 0.24.22`
    - Products have workspaces.
    - This update includes a PostgreSQL migration to add workspaces to each product.

>     - New helm chart values are added to facilitate this migration.

Reviewer: ```suggestion
    - Added new Helm chart values to facilitate the PostgreSQL migration.
```

>         - This includes a new required whitelisted API key secret used to retrieve workspace information used during the migration.

Reviewer: ```suggestion
        - This update includes a new required whitelisted API key secret. This secret is used to retrieve workspace information during the migration.
```

>         - The migration should complete quickly automatically after deployment with no required manual intervention.

Reviewer: ```suggestion
        - The migration completes automatically after deployment with no required manual intervention.
```

## New Features and Behavior changes

- Align the x-axis of your [data table](https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/visualizing-data-tables-in-a-data-space.html) plots to zero.
- Decimate the data in your [Data Spaces](https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/visualizing-data-tables-in-a-data-space.html) plots using Lossy, Max/Min, or Entry/Exit modes.
- Filter and save a custom view of a [test plans](https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/creating-a-test-plan.html) grid.
- SystemLink Enterprise adds the [workspace](https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/creating-a-workspace.html) property to products. For new products, assign a workspace so you can control access. For products already in your system, SystemLink Enterprise assigns a workspace using the following logic.

| | |
|--|--|
| The product has test results. | SystemLink Enterprise assigns the workspace associated with the first test results listed for the product. |
| The product does not have test results. | SystemLink Enterprise assigns the default workspace.|

> - `helium-serviceregistry 0.16.7`

Reviewer 2: We don't need to reference the Helm chart. We can promote the sub-bullet of this to a major bullet and remove this line.

    - The Alarm Service privileges are exposed under the `Alarms` category in the security UI.

> - `testmonitorservice 0.24.22`

Reviewer 2: Should this section be in the Helm chart breaking changes section? That section appears to be missing.

    - Products have workspaces.
    - This update includes a PostgreSQL migration to add workspaces to each product.
    - Added new Helm chart values to facilitate the PostgreSQL migration.

>         - This update includes a new required whitelisted API key secret. This secret is used to retrieve workspace information during the migration.

Reviewer 3: FYI @vpstoynova I've got [a PR up](https://dev.azure.com/ni/Users/_git/Documentation%20Reviews/pullrequest/758039) to include this in the ni.com docs as well.

>         - The migration completes automatically after deployment with no required manual intervention.

Reviewer 3: @prestwick I would move this line and the following line up a few lines to be underneath line 21. Then you can rework the new required whitelisted API key line to be more prominent, since it requires user action. Maybe put a [ACTION REQUIRED] or something?

## New Features and Behavior changes

- Align the x-axis of your [data table](https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/visualizing-data-tables-in-a-data-space.html) plots to zero.
- Decimate the data in your [Data Spaces](https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/visualizing-data-tables-in-a-data-space.html) plots using Lossy, Max/Min, or Entry/Exit modes.
- Filter and save a custom view of a [test plans](https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/creating-a-test-plan.html) grid.
- The Alarm Service privileges are exposed under the `Alarms` category in the security UI.
- SystemLink Enterprise adds the [workspace](https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/creating-a-workspace.html) property to products. For new products, assign a workspace so you can control access. For products already in your system, SystemLink Enterprise assigns a workspace using the following logic.

| | |
|--|--|

> | The product has test results. | SystemLink Enterprise assigns the workspace associated with the first test results listed for the product. |

Reviewer 3: "First" isn't as precise as we could be. Its first when ordered by started at, descending, so "most recent" is clearer.
```suggestion
| The product has test results. | SystemLink Enterprise assigns the workspace associated with the most recent test result listed for the product. |
```
